### PR TITLE
Drop Microsoft.AspNetCore.Http.Abstractions for netcoreapp3.1 and higher

### PR DESCRIPTION
**Pull request by trejjam**
## Changes

This PR handle issue with dependency restoring for .NetCore 3.1 and higher. There is no `Microsoft.AspNetCore.Http.Abstractions` after .NetCore 2.2. The `Http.Abstraction` requires other v2.x packages, which causes resolving issues for newer frameworks.

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj [hunk 1]
```diff
@@ -13,8 +13,23 @@
 
   <ItemGroup>
     <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry\OpenTelemetry.csproj" />
-    <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="$(MicrosoftAspNetCoreHttpAbstractionsPkgVer)" />
     <PackageReference Include="Microsoft.AspNetCore.Http.Features" Version="$(MicrosoftAspNetCoreHttpFeaturesPkgVer)" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj:19]: its not clear why do we have to add netcoreapp3.1 and net5.0 targets, as they are already covered by netstandard2.1 target.
For netstandard2.1, we can take frameworkreference, and for everything else, we can packagereference Microsoft.AspNetCore.Http.Abstractions

Comment (trejjam) [reply src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj:19]: I am afraid that it is not possible.

When I change the `ItemGroup` for .NetStandard2.1 like this:
```csproj
<ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.1'">
  <FrameworkReference Include="Microsoft.AspNetCore.App" />
</ItemGroup>
```

`dotnet build` returns following error:
```log
dotnet/sdk/5.0.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(326,5): error NETSDK1073: The FrameworkReference 'Microsoft.AspNetCore.App' was not recognized [../opentelemetry-dotnet/src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj]
```

**Comment by CodeBlanch** [comment]
@trejjam This is an interesting idea! I have only seen the end-of-line applications reference `Microsoft.AspNetCore.App`. Is this a common thing done by libraries? Do you have any favorite examples doing the same thing we could check out?

**Comment by trejjam** [comment]
The oldest I know are https://github.com/serilog/serilog-aspnetcore/pull/165 or https://github.com/toddams/RazorLight/blob/master/src/RazorLight/RazorLight.csproj

**Comment by cijothomas** [comment]
.NET has published guidelines on this here : https://docs.microsoft.com/en-us/aspnet/core/fundamentals/target-aspnetcore?view=aspnetcore-5.0&tabs=visual-studio

**Comment by cijothomas** [comment]
@trejjam Sorry for coming back so late. Would you have time to resolve the conflicts in this PR, so we can take this forward? the change looks good and is in alignment with that .NET guidelines.

**Comment by trejjam** [comment]
Yes, I will resolve it.

**Comment by trejjam** [comment]
@cijothomas Done

**Comment by trejjam** [comment]
Hello @cijothomas, can you review it?

**Review by cijothomas — APPROVED** [review]

**Pull request merged**